graphql_compiler/compiler/emit_sql.py

```diff
@@ -239,19 +340,152 @@ def start_global_operations(self):
 
     def filter(self, predicate):
         """Execute a Filter Block."""
-        sql_expression = predicate.to_sql(self._aliases, self._current_alias)
+        if self._in_fold or isinstance(predicate.left, FoldedContextField) \
+                or isinstance(predicate.right, FoldedContextField):
```

Comment:
Yes, I'm not sure its possible to produce such an IR because we only permit filters that take variables or values as arguments, not other predicates. A complete solution to this would probably involve traversing the left and right predicates until you get to a leaf value or variable. Implementing filters for folds will be one of the next things to do so it might be worth it to just leave the fully complete check until then.

Reply:
One other thing to keep in mind here. Use implicit line joining and avoid backticks (`\`) whenever possible, per the Google Style Guide (note our line length is 100 chars, otherwise the same rules apply): https://github.com/google/styleguide/blob/gh-pages/pyguide.md#32-line-length

Reply:
I don't know if it's possible to use `with ... as ...` blocks with `isinstance` - it returns `AttributeError: __enter__`

How is just using regular variable assignment to refer to the results of the `isinstance` calls?